### PR TITLE
Add Travis to Monero with the depends toolchain.

### PR DESCRIPTION
**Pull request by TheCharlatan**
This adds Travis CI to Monero. I do this in full awareness that there is an existing build bot, which is much more advanced than this quite simple CI config, but there are some advantages to this setup.

- Makes it transparent which dependencies and versions thereof are guaranteed to be supported.
- Cross compiles to a bunch of targets as well, like the current build bot.
- Runs the CI even on you own branches, if you activate it. This means for example that when you are developing on windows 64 bit, you do not need to open a pull request for the current build bot to tell you that on arm 32 bit, you are missing macro definitions in your patch. Travis CI will catch this, when you push to your own remote on github.

The builds for windows and mac are still failing because pcsc cannot be cross compiled. 32 bit Linux is also failing, I may need some assistance here on how to get libblocks compiled correctly for the i686 architecture. I am waiting for the hid pr to be merged to fix that as well.

### CODE REVIEW

**Comment by TheCharlatan** [comment]
The repository maintainers need to activate it on https://travis-ci.org/ , if this is a desired feature.

File: contrib/depends/packages/sodium.mk [hunk 1]
```diff
@@ -1,8 +1,8 @@
 package=sodium
-$(package)_version=1.0.16
-$(package)_download_path=https://github.com/jedisct1/libsodium/archive
-$(package)_file_name=$($(package)_version).tar.gz
-$(package)_sha256_hash=0c14604bbeab2e82a803215d65c3b6e74bb28291aaee6236d65c699ccfe1a98c
+$(package)_version=1.0.15
```
Comment (MoroccanMalinois) [inline contrib/depends/packages/sodium.mk]: Any particular reason to downgrade libsodium ?

Comment (TheCharlatan) [reply contrib/depends/packages/sodium.mk]: I could not get that version to play nice with the darwin build. If it is a problem, and there is a reason for taking 1.0.16 , I'll try to get it running.

**Comment by xiphon** [comment]
> 32 bit Linux is also failing, I may need some assistance here on how to get libblocks compiled correctly for the i686 architecture.

Working on it. Implementing architecture-independent approach.

**Pull request merged**